### PR TITLE
[1.8 release] Switch to LM dataset in torchtext 0.9.0 release

### PR DESCRIPTION
**Pull request by zhangguanheng66**
In torchtext 0.9.0 release, we will include the raw text datasets as beta release. This PR is to switch the data to the raw dataset.

This PR should be tested against pytorch 1.8.0 rc and torchtext 0.9.0 rc.

### CODE REVIEW

**Comment by zhangguanheng66** [comment]
cc @brianjo @cpuhrsch

File: beginner_source/transformer_tutorial.py [hunk 1]
```diff
@@ -145,25 +145,25 @@ def forward(self, x):
...
-from torchtext.utils import download_from_url, extract_archive
+# With torchtext 0.9.0 rc
+# from torchtext.datasets import WikiText2
+from torchtext.experimental.datasets.raw import WikiText2
 from torchtext.data.utils import get_tokenizer
 from torchtext.vocab import build_vocab_from_iterator
 
-url = 'https://s3.amazonaws.com/research.metamind.io/wikitext/wikitext-2-v1.zip'
-test_filepath, valid_filepath, train_filepath = extract_archive(download_from_url(url))
+train_iter, = WikiText2(split=('train'))
```
Comment (cpuhrsch) [inline beginner_source/transformer_tutorial.py:154]: ```('train')``` is equivalent to ```'train'```, but different from the single elemen tuple ```('train',`)))

File: beginner_source/transformer_tutorial.py [hunk 2]
```diff
@@ -145,25 +145,25 @@ def forward(self, x):
...
 tokenizer = get_tokenizer('basic_english')
-vocab = build_vocab_from_iterator(map(tokenizer,
-                                      iter(io.open(train_filepath,
-                                                   encoding="utf8"))))
+vocab = build_vocab_from_iterator(map(tokenizer, train_iter))
```
Comment (cpuhrsch) [inline beginner_source/transformer_tutorial.py:156]: To be consistent we should probably use Vocab here too

**Review by cpuhrsch — APPROVED** [review]
LGTM

**Pull request merged**